aten/src/THCUNN/generic/SparseLinear.cu

```diff
@@ -4,17 +4,17 @@
 
 static bool THNN_(checkInput)(THCTensor* t)
 {
-  return !t->is_empty() && THTensor_nDimensionLegacyAll(t) == 2 && t->size(1) == 3;
+  return !t->is_empty() && t->dim() == 2 && t->size(1) == 3;
```

Comment:
no, this is right, there are two differences between THTensor_nDimensionLegacyAll and dim:
`THTensor_nDimensionLegacyAll` collapses empty tensors to 0 (that is checked with empty here).
`THTensor_nDimensionLegacyAll` unexpands scalrs to 1 dim.  This compares to 2, so that doesn't matter.